extensions/amp-social-share/1.0/social-share-config.js

```diff
@@ -14,19 +14,23 @@
  * limitations under the License.
  */
 
-import {dict} from '../../../src/utils/object';
-
 /**
  * Get social share configurations by supported type.
- * @param  {string} type
- * @return {!SocialShareConfigDef}
+ * @param  {?string} type
+ * @return {?SocialShareConfigDef}
  */
 export function getSocialConfig(type) {
   return BUILTINS[type];
 }
 
 /**
- * @typedef {!Object<string, !JsonObject|string>}
+ * @typedef {{
+ *   shareEndpont:           string,
+ *   defaultParams:          Object<string, string>,
+ *   defaultColor:           string,
+ *   defaultBackgroundColor: string,
+ *   bindings:               ?Array<string>,
+ * }}
  */
 let SocialShareConfigDef;
```

Comment:
WDYT about moving this to `social-share.type.js` so it can share the namespace?

Reply:
I like the idea of defining more types in `social-share.type.js`.  I also like the idea of keeping it in the config file since all the data is in the config so easy to keep everything together.

I'm not sure I fully understand the implications regarding namespace, could you elaborate?

Reply:
If we put it all in the `SocialShareDef` namespace, this can be `SocialShareDef.Config`, much like we do for other Bento components i.e. `DateDisplay`. It's just a rec for consistency is all.

Reply:
Sure, updated in latest version.